doc/basics.md

```diff
@@ -67,6 +67,8 @@ Once you have an element, you can navigate it with idiomatic C++ iterators, oper
   first element.
   > Note that array[0] does not compile, because implementing [] gives the impression indexing is a
   > O(1) operation, which it is not presently in simdjson.
+* **Checking an Element Type:** You can check an element's type with `element.type()`. It
+  returns an `element_type`.
```

Comment:
Here I would be tempted to document this with a sentence such as... 

The possible values are
- simdjson::dom::element_type::ARRAY,
- simdjson::dom::element_type::OBJECT, 
- simdjson::dom::element_type::INT64, 
- simdjson::dom::element_type::UINT64, 
- simdjson::dom::element_type::DOUBLE, 
- simdjson::dom::element_type::STRING, 
- simdjson::dom::element_type::BOOL, 
- simdjson::dom::element_type::NULL_VALUE

Either that, or give a switch case example?